src/modules/person/api/person.controller.ts

```diff
@@ -438,4 +449,55 @@ export class PersonController {
 
         return { ok: true, value: personResult.value.newPassword! };
     }
+
+    @Put(':personId/lock-user')
+    @HttpCode(HttpStatus.ACCEPTED)
+    @ApiOkResponse({ description: 'User has been successfully updated.', type: PersonLockResponse })
+    @ApiNotFoundResponse({ description: 'The person was not found.' })
+    @ApiForbiddenResponse({ description: 'Insufficient permissions to perform operation.' })
+    @ApiInternalServerErrorResponse({ description: 'An internal server error occurred.' })
+    public async lockPerson(
+        @Param('personId') personId: string,
+        @Body() lockUserBodyParams: LockUserBodyParams,
+        @Permissions() permissions: PersonPermissions,
+    ): Promise<PersonLockResponse> {
+        const personResult: Result<Person<true>> = await this.personRepository.getPersonIfAllowed(
+            personId,
+            permissions,
+        );
+
+        if (!personResult.ok) {
+            throw SchulConnexErrorMapper.mapSchulConnexErrorToHttpException(
+                SchulConnexErrorMapper.mapDomainErrorToSchulConnexError(
+                    new MissingPermissionsError('Person not found or no permissions.'),
+                ),
+            );
+        }
+
+        if (!personResult.value?.keycloakUserId) {
+            throw SchulConnexErrorMapper.mapSchulConnexErrorToHttpException(
+                SchulConnexErrorMapper.mapDomainErrorToSchulConnexError(new EntityNotFoundError('Person', personId)),
+            );
+        }
+
+        const tempAttributes: Record<string, string> = {};
+        tempAttributes[LOCK_KEYS[0] as string] = lockUserBodyParams.locked_from;
+        tempAttributes[LOCK_KEYS[1] as string] = new Date().toISOString();
```

Comment:
`as` casts are not great. I'd recommend defining `LOCK_KEYS` as a tuple (`[string, string]`) then TS will know indices 0 and 1 are valid.